woocommerce/woocommerce-ios
# Shipping Labels: display carrier and rates information, and allow the selection

Part of #4085 

## Description
In this PR I continue the work started [here](https://github.com/woocommerce/woocommerce-ios/pull/4229) on the UI of Carriers and Rates.
Now:
- we will display the correct information about every carrier;
- we will display the information about extra rates of every carrier (signature or adult signature, if available);
- we will allow selecting a rate, and to add the request of a signature or adult signature;

**The unit tests of the view models are part of another PR https://github.com/woocommerce/woocommerce-ios/pull/4275**

## Changes
- I updated the `insurance` property under the `ShippingLabelCarrierRate`, since it’s a `Double`.
- Implemented `ShippingLabelCarrierRowViewModel` as a separate file.
- `ShippingLabelCarrierRow` now is able to handle the selected/unselected state, and also to display the extra info about signature and adult signature.
- `ShippingLabelCarriersViewModel` now handles the selected rates and has a new method for generating the rows.

## Testing
1. Launch the app in debug mode (because of the feature flag).
2. Open an order detail that is eligible for the shipping label.
3. Press the button "Create Shipping Label".
4. Press the button "Continue" in Create Shipping Label form, under the **Ship From** cell. Confirm the address.
5. Press the button "Continue" in Create Shipping Label form, under the **Ship To** cell. Confirm the address.
6. Press the button "Continue" in Create Shipping Label form, under the **Package Details** cell.
7. You should see the screen that shows the Package Details screen and all the items that will be shipped.
8. Press the "Package selected" row. 
9. Select a package, then press "Done".
10. Add a Total package weight, press "Done".
11. The "Shipping Carrier and Rates" cell should be enabled now. Press it.
12. You should see the new screen "Carrier and Rates". It should display first the ghost cells, then the data fetched from the API.
13. Make sure the info of every carrier is shown in the correct way, and that you are able to select a carrier and a signature/adult signature. (You will not be able to confirm your final choice, it will be implemented in the next PR).

## Screenshot
<img src="https://user-images.githubusercontent.com/495617/119490135-c84c0c00-bd5c-11eb-99d0-133ea808b07c.png" width=300 />



Update release notes:
